### PR TITLE
[VarDumper] Move $prev as 1st argument in ThrowingCasterException::__construct, removing $caster

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by stof** [comment]
:+1:

File: src/Symfony/Component/VarDumper/Exception/ThrowingCasterException.php [hunk 1]
```diff
@@ -17,11 +17,15 @@
...
     /**
-     * @param callable   $caster The failing caster
-     * @param \Exception $prev   The exception thrown from the caster
+     * @param \Exception $prev The exception thrown from the caster
      */
-    public function __construct($caster, \Exception $prev)
+    public function __construct($prev, \Exception $e = null)
     {
+        if (!$prev instanceof \Exception) {
+            @trigger_error('Providing the $prev exception as 2nd argument when instanciating a '.__CLASS__.' is deprecated since version 2.8 and will be removed in 3.0. Provide it as first argument instead.', E_USER_DEPRECATED);
```
Comment (xabbuh) [inline src/Symfony/Component/VarDumper/Exception/ThrowingCasterException.php]: Isn't the real change that you should not pass the caster anymore (which implies that the exception is now the first argument)?

Comment (Tobion) [inline src/Symfony/Component/VarDumper/Exception/ThrowingCasterException.php]: Agree, the constructor should only have one argument and the BC layer implementation should be changed.

Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Exception/ThrowingCasterException.php]: I changed the deprecation message (but not the logic around, which looks fine to me)

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**